backend/src/Controllers/elections.controllers.ts

```diff
@@ -9,14 +9,44 @@ var ElectionsModel = new ElectionsDB();
 const getElectionByID = async (req: any, res: any, next: any) => {
     console.log(`-> elections.getElectionByID ${req.params.id}`)
     try {
-        const election = await ElectionsModel.getElectionByID(parseInt(req.params.id))
+        var election = await ElectionsModel.getElectionByID(parseInt(req.params.id))
         console.log(`get election ${req.params.id}`)
-        if (!election){
+        if (!election) {
             console.log('Error')
             return res.status('400').json({
                 error: "Election not found"
             })
         }
+        // Update Election State
+        const currentTime = new Date();
+        if (election.state === 'draft') {
+
+        }
+        if (election.state === 'finalized') {
+            if (election.start_time) {
+                console.log(`-> Election Transitioning To Open From ${election.state}`)
+                const startTime = new Date(election.start_time);
+                if (currentTime.getTime() > startTime.getTime()) {
+                    election.state = 'open';
```

Comment:
Looks like we're using a lazy initiazation approach to not update the election state until someone retrieves it? I like that approach for a lot of scenarios, but we may hit some issues with this particular case

1. The getElectionByID api name implies that it's only doing reads, but it performs writes as an unintended side effect
2. In frequent calls to getElectionByID would leave our database in an incorrect state (showing the finalized state when the election should really be in the open state), and I could see this leading to other bugs

Here are some options for fixing it, they're kind of tricky so this is definitely a task for later
 * Setup a process to update the database regardless of the customer traffic. Maybe a cron job? I'm not sure how this would work on heroku
 * Refactor our states to avoid time sensitive states. Here ``finalized``, ``open``, and ``closed`` would all be stored as ``finalized`` in the database, and we'd determine pending/open/closed on the fly based on the time

Reply:
Yea I've been trying to figure out the best way to handle this for a while. I decided to just do the lazy check here and open an issue later for handling it better. There's a library node-cron that can handle timed events but I wasn't sure the best way to configure it in an express app. I was thinking every 5 minutes it could run a query for all finalized or open elections, update the state if needed and send any required emails. That could also be a good time to update election results rather than calculating them every time someone goes to the results page.